source/lib.ts

```diff
@@ -0,0 +1,264 @@
+// /source/lib.ts
+// The option parser and rate limiting middleware
+
+import Express from 'express'
+
+import { Options, AugmentedRequest, RateLimitRequestHandler } from './types.js'
+import MemoryStore from './memory-store.js'
+
+/**
+ * Adds the defaults for options the user has not specified.
+ *
+ * @param options {Options} - The options the user specifies
+ *
+ * @returns {Options} - A complete configuration object
+ */
+const parseOptions = (passedOptions: Partial<Options>): Options => {
+	// Now add the defaults for the other options
+	const options: Options = {
+		windowMs: 60 * 1000,
+		store: new MemoryStore(passedOptions.windowMs ?? 60 * 1000),
```

Comment:
I think we should stop passing anything into the constructor, and instead add an `init(config, limiter)` method to the store API - that way third-party stores would have an official way of getting the config instead of the current solution of requiring end-users to pass in the values twice. We'd want to make the init method optional and only call it if it exists.

Reply:
I'm curious, how would the stores use the middleware passed as the `limiter` param in the `init` function?